ui/src/app/pages/admin/admin-user-common.tsx

```diff
@@ -0,0 +1,325 @@
+    contactEmail: getUpdatedProfileValue(oldProfile, updatedProfile, 'contactEmail'),
+    affiliation: getUpdatedProfileValue(oldProfile, updatedProfile, 'verifiedInstitutionalAffiliation'),
+    accessBypassRequests: [],  // coming soon: RW-4958
+  };
+
+  return userAdminApi().updateAccountProperties(request);
+}
+
+export const DropdownWithLabel =
+  ({label, options, initialValue, onChange, disabled= false, dataTestId, labelStyle = {}, dropdownStyle = {}}) => {
+  return <FlexColumn data-test-id={dataTestId} style={{marginTop: '1rem'}}>
+    <label style={{...commonStyles.label, ...labelStyle}}>{label}</label>
+    <Dropdown
+      style={{...commonStyles.dropdown, ...dropdownStyle}}
+      options={options}
+      onChange={(e) => onChange(e)}
+      value={initialValue}
+      disabled={disabled}
+    />
+  </FlexColumn>;
+};
+
+interface ContactEmailTextInputProps {
+  contactEmail: string,
+  onChange: Function,
+  labelStyle?: CSSProperties,
+  inputStyle?: CSSProperties,
+  containerStyle?: CSSProperties,
+}
+export const ContactEmailTextInput = ({contactEmail, onChange, labelStyle, inputStyle, containerStyle}: ContactEmailTextInputProps) => {
+  return <TextInputWithLabel
+    inputId={'contactEmail'}
+    labelText={'Contact email'}
+    value={contactEmail}
+    labelStyle={{...commonStyles.label, ...labelStyle}}
+    inputStyle={{...commonStyles.textInput, ...inputStyle}}
+    containerStyle={{...commonStyles.textInputContainer, ...containerStyle}}
+    onChange={value => onChange(value)}/>;
+}
+
+interface FreeCreditsDropdownProps {
+  initialLimit?: number,
+  currentLimit?: number,
+  onChange: Function,
+  labelStyle?: CSSProperties,
+  dropdownStyle?: CSSProperties,
+}
+export const FreeCreditsDropdown = ({initialLimit, currentLimit, onChange, labelStyle, dropdownStyle}: FreeCreditsDropdownProps) => {
+  return <DropdownWithLabel
+    dataTestId={'freeTierDollarQuota'}
+    label={'Free credit limit'}
+    options={getFreeCreditLimitOptions(initialLimit)}
+    initialValue={currentLimit}
+    labelStyle={labelStyle}
+    dropdownStyle={dropdownStyle}
+    onChange={(event) => onChange(event)}/>;
+}
+
+interface InstitutionDropdownProps {
+  institutions?: PublicInstitutionDetails[],
+  initialInstitutionShortName?: string,
+  onChange: Function,
+  labelStyle?: CSSProperties,
+  dropdownStyle?: CSSProperties,
+}
+export const InstitutionDropdown =
+  ({institutions, initialInstitutionShortName, onChange, labelStyle, dropdownStyle}: InstitutionDropdownProps) => {
+  const options = fp.map(({displayName, shortName}) => ({label: displayName, value: shortName}), institutions);
+  return institutions
+    ? <DropdownWithLabel
+      dataTestId={'verifiedInstitution'}
+      label={'Verified institution'}
+      options={options}
+      initialValue={initialInstitutionShortName}
+      labelStyle={labelStyle}
+      dropdownStyle={dropdownStyle}
+      onChange={(event) => onChange(event)}/>
+    : null;
```

Comment:
should there be a spinner or error message if we cannot get institutions list

Reply:
We have this on initial load.  Because UserAdminProfilePage is rendered with `withRoutingSpinner`, it has a spinner until we turn it off.  This happens on line 169, after we populate institutions.

It's true that errors in the institution response are not handled - do you think that's important to have here?

Reply:
So now that i am thinking about it,there are two scenarios:
1) The user has institution assigned: We get the value of the institution user belong to, has but not the complete list. : Thats ok we can at least show the user the assigned institution and thats it
2) User doesnt have institution assigned (maybe old users): Not sure how many of these exist and if they can access anything. But in such cases if we cannot show the institution list that will be an issue and we would need to show an error message

Reply:
I'll add this to the TODOs in the description